### DISCUSSION TITLE
Errors while running SimpleExample.ipynb

### DISCUSSION THREAD

**Discussion by somyagr** [ModECI/MDF#222] [Q&A] [answered]
Hi mentors,
Even after installing all the dependencies by running `pip install .` ,  I am getting many errors while running SimpleExample.ipynb in jupyter notebook like "No module named 'onnx' ". After installing onnx by pip command, I got the next error "No module named 'modelspec' " and since then there have been a lot of errors. Please tell me if I'm going wrong or I'll have to install these modules like this only. 
Thanks in advance.

**Comment by pgleeson** [comment, marked as answer]
@somyagr What OS are you using and have you set up a virtual environment? Can you paste here the output of `pip list`?

**Reply by somyagr** [reply]
I am using Ubuntu 20.04, yes I have set up a virtual environment.

(MDF-env) HP-348-G4:~/MDF`$ pip list`
Package              Version
-------------------- --------
airspeed             0.5.19
argon2-cffi          21.3.0
argon2-cffi-bindings 21.2.0
asttokens            2.0.5
attrs                21.4.0
autograd             1.3
backcall             0.2.0
beautifulsoup4       4.10.0
bleach               4.1.0
bokeh                2.4.2
boltons              21.0.0
cachetools           5.0.0
cattrs               1.10.0
cffi                 1.15.0
click                8.1.0
cloudpickle          2.0.0
colorama             0.4.4
commonmark           0.9.1
cycler               0.11.0
Cython               0.29.28
dask                 2022.3.0
debugpy              1.6.0
decorator            4.4.2
defusedxml           0.7.1
dill                 0.3.4
distributed          2022.3.0
docstring-parser     0.13
elfi                 0.8.2
entrypoints          0.4
executing            0.8.3
face                 20.1.1
fsspec               2022.2.0
future               0.18.2
glom                 22.1.0
GPy                  1.10.0
graph-scheduler      1.0.0
graphviz             0.16
grpcio               1.42.0
grpcio-tools         1.42.0
h5py                 3.6.0
HeapDict             1.0.1
importlib-resources  5.6.0
ipykernel            6.10.0
ipyparallel          8.2.0
ipython              8.2.0
ipython-genutils     0.2.0
ipywidgets           7.7.0
jedi                 0.18.1
Jinja2               3.1.1
joblib               1.1.0
jsonschema           4.4.0
jupyter              1.0.0
jupyter-client       7.2.0
jupyter-console      6.4.3
jupyter-core         4.9.2
jupyterlab-pygments  0.1.2
jupyterlab-widgets   1.1.0
kiwisolver           1.4.2
leabra-psyneulink    0.3.2
libNeuroML           0.3.1
llvmlite             0.37.0
locket               0.2.1
lxml                 4.8.0
MarkupSafe           2.1.1
matplotlib           3.3.3
matplotlib-inline    0.1.3
mistune              0.8.4
modeci-mdf           0.3.3
modelspec            0.2.1
mpmath               1.2.1
msgpack              1.0.3
nbclient             0.5.13
nbconvert            6.4.5
nbformat             5.2.0
nest-asyncio         1.5.4
networkx             2.5.1
neuromllite          0.4.3
notebook             6.4.10
numpy                1.20.2
onnx                 1.9.0
onnxconverter-common 1.8.1
onnxruntime          1.7.0
packaging            21.3
pandocfilters        1.5.0
paramz               0.9.5
parso                0.8.3
partd                1.2.0
pexpect              4.8.0
pickleshare          0.7.5
Pillow               8.4.0
Pint                 0.17
pip                  22.0.4
prometheus-client    0.13.1
prompt-toolkit       3.0.28
protobuf             3.19.4
psutil               5.9.0
psyneulink           0.10.0.0
ptyprocess           0.7.0
pure-eval            0.2.2
pycparser            2.21
Pygments             2.11.2
PyLEMS               0.5.9
pyNeuroML            0.5.20
pyparsing            2.4.7
pyrsistent           0.18.1
python-dateutil      2.8.2
PyYAML               6.0
pyzmq                22.3.0
qtconsole            5.3.0
QtPy                 2.0.1
rich                 10.12.0
scikit-learn         1.0.2
scipy                1.8.0
Send2Trash           1.8.0
setuptools           52.0.0
six                  1.16.0
skl2onnx             1.8.0
sortedcontainers     2.4.0
soupsieve            2.3.1
stack-data           0.2.0
sympy                1.8
tabulate             0.8.9
tblib                1.7.0
terminado            0.13.3
testpath             0.6.0
threadpoolctl        3.1.0
toolz                0.11.2
toposort             1.7
torch                1.8.0
tornado              6.1
tqdm                 4.63.1
traitlets            5.1.1
typecheck-decorator  1.2
typing_extensions    4.1.1
wcwidth              0.2.5
webencodings         0.5.1
wheel                0.36.2
widgetsnbextension   3.6.0
zict                 2.1.0
zipp                 3.7.0

**Reply by sanjayankur31** [reply]
Are you running the jupyter notebook in the virtualenv you created and installed the packages in?

**Reply by pgleeson** [reply]
Try creating a new notebook and typing `!pip list` into the first cell and make sure that the package list is the same as you get at the command line.

**Reply by somyagr** [reply]
@sanjayankur31 yes I'm running the jupyter notebook in the virtual environment only,
@pgleeson I've verified, the list is the same at both the places

**Reply by pgleeson** [reply]
Sorry @somyagr, not sure what else to suggest... If the pip list is stating that onnx is installed, but running the notebook complains that it's not, it's an inconsistency. 

All I could suggest is making a new virtual env (or use conda to manage a fresh environment), test that the examples at https://github.com/ModECI/MDF/tree/main/examples/MDF all run fine at the command line, make sure you're using executables inside the env, e.g run `which jupyter-notebook`, and try the notebook again.

**Comment by somyagr** [comment]
@pgleeson Sorry to disturb you again for this issue, please help me out if you have any alternative.
I've installed modelspec through pip but I'm getting `No module named 'modelspec.BaseTypes'` error. Version of modelspec installed currently is 0.2.3.
I've tried uninstalling it and reinstalling via pip3 but that makes no difference.
It seems the problem is not with installing modelspec, but with the BaseTypes associated with it.
If anyone else has gone through this error please help me out.

**Reply by pgleeson** [reply]
@somyagr I would strongly recommend using conda and make a new virtual environment, basing it on python 3.9, as that's the best tested. Then install MDF, from the development branch, as that has the latest updates. It should pick up the relevant dependencies from there, inlcuding modelspec.
